// @oracle/oraclejet-preact/cjs/Flex-ca3ece5b.js
/* @oracle/oraclejet-preact: 15.1.3 */
'use strict';

var jsxRuntime = require('preact/jsx-runtime');
var utils_UNSAFE_interpolations_dimensions = require('./utils/UNSAFE_interpolations/dimensions.js');
var utils_UNSAFE_mergeInterpolations = require('./utils/UNSAFE_mergeInterpolations.js');

var utils_UNSAFE_interpolations_boxalignment = require('./utils/UNSAFE_interpolations/boxalignment.js');
require('./utils/UNSAFE_interpolations/flexbox.js');
require('./utils/UNSAFE_interpolations/flexitem.js');
require('./utils/UNSAFE_classNames.js');
var flexbox = require('./flexbox-4ee1d134.js');
var flexitem = require('./flexitem-dcfe9df9.js');
var classNames = require('./classNames-58437073.js');

const baseStyles = "_14hrzjm";
// exported for testing
// TODO: This will be removed when converting the Flex commponent to the new theming,
// and there the marginZeroStyles class name will be exported in the FlexStyles.css.ts file
// and can be used for testing.
const marginZeroStyles = "akyxpl";
const interpolations = [...Object.values(utils_UNSAFE_interpolations_dimensions.dimensionInterpolations), ...Object.values(flexbox.flexboxInterpolations), ...Object.values(flexitem.flexitemInterpolations), ...Object.values(utils_UNSAFE_interpolations_boxalignment.boxAlignmentInterpolations)];
const styleInterpolations = utils_UNSAFE_mergeInterpolations.mergeInterpolations(interpolations);
const Flex = ({
  children,
  hasZeroMargins = true,
  ...props
}) => {
  const flexComponentClasses = classNames.classNames([baseStyles, hasZeroMargins && marginZeroStyles]);
  const {
    class: cls,
    ...styles
  } = styleInterpolations(props);
  return jsxRuntime.jsx("div", {
    class: `${flexComponentClasses} ${cls}`,
    style: styles,
    children: children
  });
};

exports.Flex = Flex;
exports.marginZeroStyles = marginZeroStyles;
