// @oracle/oraclejet-preact/es/ComponentMessage-e9198bcf.js
/* @oracle/oraclejet-preact: 15.1.3 */
import { jsxs, jsx } from 'preact/jsx-runtime';
import { useComponentTheme } from './hooks/UNSAFE_useComponentTheme.js';
import { useTranslationBundle } from './hooks/UNSAFE_useTranslationBundle.js';
import './PRIVATE_Message.js';
import './UNSAFE_HiddenAccessible.js';
import { ComponentMessageRedwoodTheme } from './UNSAFE_ComponentMessage/themes/redwood/ComponentMessageTheme.js';
import { H as HiddenAccessible } from './HiddenAccessible-ce8ac79b.js';
import { i as isSeverityIconNeeded } from './MessageUtils-8e4ce279.js';
import { M as MessageStartIcon } from './MessageStartIcon-24742f3b.js';
import { M as MessageSummary } from './MessageSummary-c69260d0.js';

/**
 * The component that renders an individual message for inline component messaging.
 */
function ComponentMessage({ detail, fieldLabel, severity = 'error' }) {
    const { classes } = useComponentTheme(ComponentMessageRedwoodTheme, {
        severity: severity ?? 'none'
    });
    const translations = useTranslationBundle('@oracle/oraclejet-preact');
    const severityTranslations = {
        confirmation: translations.confirmation(),
        error: translations.error(),
        info: translations.info(),
        warning: translations.warn()
    };
    return (jsxs("div", { class: classes, role: "alert", "aria-atomic": "true", children: [fieldLabel && jsx(HiddenAccessible, { children: fieldLabel }), isSeverityIconNeeded(severity) && (jsx(MessageStartIcon, { variant: "inline", severity: severity, translations: severityTranslations })), jsx(MessageSummary, { variant: "inline", text: detail })] }));
}

export { ComponentMessage as C };
