// @oracle/oraclejet-preact/es/ComponentMessageContainer-a7ea8c1d.js
/* @oracle/oraclejet-preact: 15.1.3 */
import { jsx } from 'preact/jsx-runtime';
import './PRIVATE_Message.js';
import { C as ComponentMessage } from './ComponentMessage-e9198bcf.js';
import { componentMessageContainerBase } from './UNSAFE_ComponentMessage/themes/ComponentMessageStyles.css.js';
import { M as MessagesManager } from './MessagesManager-5a453b04.js';

/**
 * Converts the messages data into Item
 *
 * @param messages The messages data passed down as prop
 * @returns The messages data as Item
 */
function generateMessagesData(messages) {
    return messages.map((message, index) => {
        return {
            key: index,
            data: {
                closeAffordance: 'off',
                severity: message.severity || 'error',
                detail: message.detail
            }
        };
    });
}
function ComponentMessageContainer({ fieldLabel, messages = [] }) {
    return (jsx("div", { class: componentMessageContainerBase, children: jsx(MessagesManager, { data: generateMessagesData(messages), children: ({ item }) => (jsx(ComponentMessage, { detail: item.data.detail, fieldLabel: fieldLabel, severity: item.data.severity }, item.key)) }) }));
}

export { ComponentMessageContainer as C };
